storage/rocksdb/ha_rocksdb.cc

```diff
@@ -2449,6 +2453,10 @@ class Rdb_transaction_impl : public Rdb_transaction {
   rocksdb::Transaction *m_rocksdb_tx = nullptr;
   rocksdb::Transaction *m_rocksdb_reuse_tx = nullptr;
 
+  // Number of savepoints to roll back to get to the start of the
+  // "top-level" statement.
+  int m_n_savepoints;
```

Comment:
It would probably make more sense to have these variables and collapse_savepoint() call in Rdb_transaction instead?